### ISSUE TITLE
Message ID can crash the io loop.

### ISSUE THREAD

**Issue by hadrienk** [PelionIoT/java-coap#29]
In the method `DatagramSocketTransport`#`readingLoop()` catches the IllegalArgumentException thrown by `setMessageId` outside of the loop, effectively crashing the whole server.

**Comment by hadrienk** [comment]
Pretty easy to reproduce using #16: 

```
var b = new byte[]{ 0x4a, (byte) 0xab};
var invalidPacket = CoapPacket.deserialize(remoteAddress, new ByteArrayInputStream(b));

// Returns -1
invalidPacket.getMessageId() 
```

**Comment by szysas** [comment]
Hi @hadrienk,

What version did you use to try reproduce this issue. At least on master build it does not fail. 
This might have been fix by this commit: https://github.com/ARMmbed/java-coap/commit/51f9ae316c76baac805681a013f050394a957dce

**Comment by hadrienk** [comment]
I see we were using version 5.0.0. Is this commit merged in 5.1.0?

**Comment by szysas** [comment]
There is no 5.1.0 version release. I suggest to use `master` or `9cd9e8141b` as a version number.
